### ISSUE TITLE
scrcpy not working with X23

### ISSUE THREAD

**Issue by franco-bez** [HelloVolla/volla-os-beta-test#199]
there are several reports that scrcpy is not working on the X23, at least VOS 14 is affected, maybe older versions as well.
Other devices like the V22 and the quintus are not affected, it seems to be a X23 only problem.

```
scrcpy 3.0.2 <https://github.com/Genymobile/scrcpy>
INFO: ADB device found:
INFO:     -->   (usb)  GX4VCTR                   device  Volla_Phone_X23
/usr/local/share/scrcpy/scrcpy-server: 1 file pushed, 0 skipped. 150.3 MB/s (90396 bytes in 0.001s)
[server] INFO: Device: [Volla] volla Volla Phone X23 (Android 14)
INFO: Renderer: opengl
INFO: OpenGL version: 4.6 (Compatibility Profile) Mesa 24.0.9-0ubuntu0.2
INFO: Trilinear filtering enabled
INFO: Texture: 720x1560
[server] ERROR: Capture/encoding error: java.lang.IllegalArgumentException: 
[server] INFO: Retrying with -m1280...
[server] ERROR: Capture/encoding error: java.lang.IllegalArgumentException: 
[server] INFO: Retrying with -m1024...
[server] ERROR: Capture/encoding error: java.lang.IllegalArgumentException: 
[server] INFO: Retrying with -m800...
[server] ERROR: Capture/encoding error: java.lang.IllegalArgumentException: 
WARN: Device disconnected
[server] ERROR: Exception on thread Thread[video,5,main]
java.lang.IllegalArgumentException: 
 at android.media.MediaCodec.native_configure(Native Method)
 at android.media.MediaCodec.configure(MediaCodec.java:2386)
 at android.media.MediaCodec.configure(MediaCodec.java:2302)
 at com.genymobile.scrcpy.video.SurfaceEncoder.streamCapture(SurfaceEncoder.java:92)
 at com.genymobile.scrcpy.video.SurfaceEncoder.lambda$start$0$com-genymobile-scrcpy-video-SurfaceEncoder(SurfaceEncoder.java:296)
 at com.genymobile.scrcpy.video.SurfaceEncoder$$ExternalSyntheticLambda0.run(D8$$SyntheticClass:0)
 at java.lang.Thread.run(Thread.java:1012)
``` 

see also the TG groups
https://t.me/c/1289127708/4896
https://t.me/hello_volla/129136

**Comment by Achim242** [comment]
I can confirm that. I already had the problem last year in November on the X23. That must have been under VOS 12 or 13. 
It works fine with other devices. Then, as now.

**Comment by wurzer** [comment]
I suggest to open a ticket here: https://github.com/Genymobile/scrcpy/issues

A developer of this should easily be able to solve this clear error message. Looks liek an econding error. 

Looks like the arguments of this method are wrong: 

`mediaCodec.configure(format, null, null, MediaCodec.CONFIGURE_FLAG_ENCODE);`

**Comment by jasper-bk** [comment]
I also experienced this and it does not seem to be a problem with the scrcpy. I suppose it has something to do with the general codec problem I have faced in the latest stable release of the X23. Because my phone is also rebooting if some videos start to run. Sometime only with sound and sometimes with sound. I suppose its codec problem in the X23

**Comment by franco-bez** [comment]
Fixed in latest VollaOS14 release

**Comment by Achim242** [comment]
> Fixed in latest VollaOS14 release

Confirm that. Fixed in 14-20250302-STABLE-vidofnir.
